R/format_GLA.R

```diff
@@ -0,0 +1,638 @@
+
+    rr_data <- rr_data %>%
+      dplyr::filter(.data$PopID %in% pop_filter & !(is.na(.data$PopID)))
+
+  }
+
+
+  #### BROOD DATA
+  message("Compiling brood information...")
+  Brood_data <- create_brood_GLA(data)
+
+  #### CAPTURE DATA
+  message("Compiling capture information...")
+  Capture_data <- create_capture_GLA(data)
+
+  #### INDIVIDUAL DATA
+  message("Compiling individual information...")
+  Individual_data <- create_individual_GLA(Capture_data)
+
+  #### LOCATION DATA
+  message("Compiling location information...")
+  Location_data <- create_location_GLA(data)
+
+  time <- difftime(Sys.time(), start_time, units = "sec")
+
+  message(paste0("All tables generated in ", round(time, 2), " seconds"))
+
+  #### EXPORT DATA
+
+  if(output_type == "csv"){
+
+    message("Saving .csv files...")
+
+    utils::write.csv(x = Brood_data, file = paste0(path, "\\Brood_data_GLA.csv"), row.names = F)
+
+    utils::write.csv(x = Capture_data, file = paste0(path, "\\Capture_data_GLA.csv"), row.names = F)
+
+    utils::write.csv(x = Individual_data, file = paste0(path, "\\Individual_data_GLA.csv"), row.names = F)
+
+    utils::write.csv(x = Location_data, file = paste0(path, "\\Location_data_GLA.csv"), row.names = F)
+
+    invisible(NULL)
+
+  }
+
+  if(output_type == "R"){
+
+    message("Returning R objects...")
+
+    return(list(Brood_data = Brood_data,
+                Capture_data = Capture_data,
+                Individual_data = Individual_data,
+                Location_data = Location_data))
+
+  }
+
+}
+
+#### --------------------------------------------------------------------------~
+#### FUNCTIONS
+#### --------------------------------------------------------------------------~
+
+
+#' Create brood data table for great tits and blue tits in Glasgow, Scotland.
+#'
+#' Create brood data table in standard format for Glasgow, Scotland.
+#'
+#' @param data Data frame of modified primary data from Glasgow, Scotland.
+#'
+#' @return A data frame.
+
+create_brood_GLA <- function(nest_data, rr_data) {
+
+  ## Create brood data template
+  load(file = "data/Brood_data_template.rda")
+
+  ## Get brood data from ringing records
+  rr_data_brood_sum <- rr_data %>%
+    dplyr::filter(is.na(.data$LocationID) == F) %>%
+
+    ## Determine whether a chick or adult - done for each observation since there is no unique identifier for individuals
+    dplyr::mutate(RingAge = ifelse(.data$Age_observed == 1, "chick", "adult")) %>%
+
+    ## Keeping chicks
+    dplyr::filter(RingAge == "chick") %>%
+
+    ## Summarize brood information for each nest
+    dplyr::group_by(.data$BreedingSeason, .data$PopID, .data$LocationID) %>%
+    dplyr::summarise(Species = names(which.max(table(.data$Species, useNA = "always"))),
+                     FemaleID = names(which.max(table(.data$MotherRing, useNA = "always"))),
+                     MaleID = names(which.max(table(.data$FatherRing, useNA = "always"))),
```

Comment:
There are not (double-checked just summarizing by unique values), but I thought in case band numbers were written down incorrectly, then taking the most common name wouldn't break and would be more likely to return the right value. 
Perhaps this is a case though where it is better to get an error so that we know there is an issue?

Reply:
I think throwing an error would be better so that we can detect it if it happens. Could occur due to a bug or be an issue from the data. Either way would be good to know.